contrib/dropwizard/src/main/java/io/opencensus/contrib/dropwizard/DropWizardMetrics.java

```diff
@@ -0,0 +1,339 @@
+import io.opencensus.metrics.export.MetricDescriptor;
+import io.opencensus.metrics.export.MetricDescriptor.Type;
+import io.opencensus.metrics.export.MetricProducer;
+import io.opencensus.metrics.export.Point;
+import io.opencensus.metrics.export.Summary;
+import io.opencensus.metrics.export.Summary.Snapshot;
+import io.opencensus.metrics.export.Summary.Snapshot.ValueAtPercentile;
+import io.opencensus.metrics.export.TimeSeries;
+import io.opencensus.metrics.export.Value;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map.Entry;
+import javax.annotation.Nullable;
+
+/**
+ * Collects DropWizard metrics from a MetricRegistry.
+ *
+ * <p>A {@link io.opencensus.metrics.export.MetricProducer} that wraps a DropWizardMetrics {@link
+ * com.codahale.metrics.MetricRegistry}.
+ *
+ * @since 0.17
+ */
+public class DropWizardMetrics extends MetricProducer {
+
+  @DefaultVisibilityForTesting static final String DEFAULT_UNIT = "1";
+
+  @DefaultVisibilityForTesting
+  static final LabelKey RATE_LABEL_KEY =
+      LabelKey.create("rate", "measures the rate of events over time");
+
+  @DefaultVisibilityForTesting
+  static final LabelValue RATE_MEAN_LABEL_VALUE = LabelValue.create("mean_rate");
+
+  @DefaultVisibilityForTesting
+  static final LabelValue RATE_ONE_MINUTE_LABEL_VALUE = LabelValue.create("m1_rate");
+
+  @DefaultVisibilityForTesting
+  static final LabelValue RATE_FIVE_MINUTE_LABEL_VALUE = LabelValue.create("m5_rate");
+
+  @DefaultVisibilityForTesting
+  static final LabelValue RATE_FIFTEEN_MINUTE_LABEL_VALUE = LabelValue.create("m15_rate");
+
+  private final com.codahale.metrics.MetricRegistry metricRegistry;
+  private final Clock clock;
+  private final Timestamp cumulativeStartTimestamp;
+
+  /**
+   * Hook the Dropwizard registry into the OpenCensus.
+   *
+   * @param metricRegistry the instance of the {@link com.codahale.metrics.MetricRegistry}.
+   * @since 0.17
+   */
+  public DropWizardMetrics(com.codahale.metrics.MetricRegistry metricRegistry) {
+    Utils.checkNotNull(metricRegistry, "metricRegistry");
+    this.metricRegistry = metricRegistry;
+    clock = MillisClock.getInstance();
+    cumulativeStartTimestamp = clock.now();
+  }
+
+  /**
+   * Returns a {@code Metric} collected from {@link Gauge}.
+   *
+   * @param dropwizardName the metric name.
+   * @param gauge the gauge object to collect.
+   * @return a {@code Metric}.
+   */
+  @SuppressWarnings("rawtypes")
+  private @Nullable Metric collectGauge(String dropwizardName, Gauge gauge) {
+    String metricName = DropWizardUtils.generateFullMetricName(dropwizardName, "gauge");
+    String metricDescription = DropWizardUtils.generateFullMetricDescription(dropwizardName, gauge);
+
+    // Figure out which gauge instance and call the right method to get value
+    Type type;
+    Value value;
+
+    Object obj = gauge.getValue();
+    if (obj instanceof Long
```

Comment:
My feeling is that we try to be a bit too clever here. What about always use Double value for any Number. Reason is that if the dorpwizard gauge uses AtomicLong or DoubleAdder should be fine.

Reply:
Done